bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/SingleDirectoryDbLedgerStorage.java

```diff
@@ -829,7 +831,7 @@ private void swapWriteCache() {
     public void flush() throws IOException {
         Checkpoint cp = checkpointSource.newCheckpoint();
         checkpoint(cp);
-        checkpointSource.checkpointComplete(cp, true);
+        checkpointSource.checkpointComplete(cp, true, ledgerDirsManager);
```

Comment:
@hangc0276 
SingleDirectoryDbLedgerStorage's ledgerDirsManager can't cover all directories, only currently managed directory,I think you need all ledger' dirs 
<img width="200" alt="image" src="https://user-images.githubusercontent.com/42990025/180979609-ce424961-0d19-4e90-bddb-e29af2af6bbe.png">

so the structure of LedgerDirsManager is very confusing, I am also optimizing this, see for details in mail talking with @eolivelli 
https://lists.apache.org/thread/njw4yhdgv9db64notq5o3hc0k8bbpl8v